spinn_front_end_common/interface/abstract_spinnaker_base.py

```diff
@@ -1849,6 +1849,33 @@ def _end_of_run_timing(self):
             self._mapping_time, self._dsg_time, self._load_time,
             self._execute_time, self._extraction_time)
 
+    def _sort_out_provenance_writing(self, executor):
```

Comment:
Not just _write_provenance?

Reply:
well coz there already is a _write_provenance that that method calls. lol. But will look into merging the two

Reply:
nah, it makes _write_provenance over more than 1 screen worth of a method

ill rename to gather_provenance_for_writing